info_encoding.md

```diff
@@ -1,116 +1,108 @@
 
-To get a simpler encoding, we do not require schemalessness.
-Therefore, derivation steps need to define a schema for the encoding
-of the info field.
-For the core derivations, this is given.
-Custom applications need to provide this as well.
-Note that schemalessness would not help in this context, because 
-all users that are derive a key are on the same page what that key
-is supposed to be used for.
-If they are not, the adversary got them to execute their code already,
-which means that particular user is in trouble anyways (REPHRASE).
+ Note that schemalessness would not help in this context, because all users who derive a key agree, by following the specification, on what that key is supposed to be used for up front, so even in a schema
 
-We require the format to be able to handle buffers that are less than 2<sup>16</sup>B in size.
+We require the format to be able to handle buffers that are at most 65535 bytes in size.
 
-At this point, we do not require to be able to encode nested data.
-However, we will later propose two options for backwards-compatible
-changes to this proposal that support that.
+At this point, we do not require to be able to encode nested data. However, we will later propose two options for backwards-compatible changes to this proposal that support that.
 
 ## Format
 
-The proposed format is called _shallow length prefixed (SLP)_ encoding.
-Shallow, because it does not specify nesting.
-Length-prefixed, because each element is prefixed with its length.
+The proposed format is called _shallow length prefixed (SLP)_ encoding. Shallow, because it does not specify nesting. Length-prefixed, because each element is prefixed with its length.
 
-Specifically, the encoding is the element-wise concatenation of the the
-length of the elements and the element.
-The length is encoded as big endian uint16. (REALLY BIG ENDIAN? PRO/CON?)
-The pseudocode of the algorithm can be found in section [Code].
+Specifically, the encoding is the element-wise concatenation of the the length of the elements and the element. The length is a uint16 encoded using common byte order. The pseudocode of the algorithm can be found in section [Code].
 
 For example, the list
 ```
 (a, b, c)
 ```
 is encoded as:
 ```
-uint16BE(len(a)) || a || uint16BE(len(b)) || b || uint16BE(len(c)) || c
+encodeU16(len(a)) || a || encodeU16(len(b)) || b || encodeU16(len(c)) || c
+```
+
+## Ordered Key-Value Datasets
+
+Ordered key-value datasets can be encoded by first constructing and then encoding a list from the dataset.  To construct the list, start with an empty list, iterate over the dataset, and then for each key-value pair in the dataset first add the key and then the value to the list.
+
+### Example
+
+The list representation of the dataset
+
+```json
+{
+	"key1": "value1",
+	"key2": "value2"
+}
+```
+is given by
+```
+("key1", "value1", "key2", "value2")
 ```
 
 ## Nested Data
 
```

Comment:
This whole section needs to be right at the end.
We're not using it yet, we 100% don't want people to get confused by this as opposed to the thing they need right now.

I'm glad you wrote the ideas down though, they'll be useful in future perhaps